src/prefect/client/collections.py

```diff
@@ -29,6 +29,6 @@ def get_collections_metadata_client(
     """
     orchestration_client = get_client(httpx_settings=httpx_settings)
     if orchestration_client.server_type == ServerType.CLOUD:
-        return get_cloud_client(httpx_settings=httpx_settings)
+        return get_cloud_client(httpx_settings=httpx_settings, infer_cloud_url=True)
```

Comment:
this should force the client's base url to use `PREFECT_API_URL.value()` which mirrors prior behavior wherein `CloudClient.read_worker_metadata` used to reach in and use `PREFECT_API_URL` over `PREFECT_CLOUD_API_URL`. 

I made these changes because I updated the CloudClient to parse out account/workspace ids from the profile on instantiation rather than only in individual methods and so to maintain prior behavior here, this flag should cause the client to use the same setting as before. 

I considered trying to disentangle the PREFECT_API_URL vs PREFECT_CLOUD_API_URL b/c it seemed really confusing particularly this `infer_cloud_url` flag in `get_cloud_client` but I shy'd away for now and just made these changes.